### ISSUE TITLE
Performance comparison with other packages

### ISSUE THREAD

**Issue by nadeemja** [maxnowack/react-tracker-connect#13]
Hi! 

This looks promising. 

How is performance relative to react-komposer2, createContainer etc?

**Comment by maxnowack** [comment]
I developed this package because I wasn't satisfied with the performance of the other solutions.
The main goal is to rerender as less as possible. One feature, which this package but currently no other package has, is that the reactive function is only executed, if the props that are used in this function have changed.

Example: 
````es6
import React from 'react'
import connect from 'react-tracker-connect'

function Post({ title, body, backgroundColor }) {
  return (
    <div style={{ backgroundColor }}>
      <h2>{ title }</h2>
      <p>{ body }</p>
    </div>
  )
}

export default connect(({ postId }) => {
  const { title, body } = Posts.findOne({ _id: postId }, { fields: { title: 1, body: 1 } })
return { title, body }
})(Post)

// Usage
<Post backgroundColor="#fff" postId={ ... } />
````

With react-tracker-connect, the reactive function will only rerun if the postId or a reactive dependency, used in this function, has changed (minimongo in this case).
If you want to change another prop very frequently, which isn't used in the reactive function, it will automatically cache the props and don't rerun. If you change the background color every 100ms for example, the minimongo query would only be executed, if the postId, or the title or body of the post has changed. With all other packages, the query will be unnecessarily rerun.
This doesn't really matter, if only a few posts are displayed. But imagine you try to render > 500 posts on a single page.

Additionally you can finetune the performance with some [options](https://github.com/maxnowack/react-tracker-connect#options)

I'll leave this issue open for discussing performance of the several tracker-react packages out there.

**Comment by nadeemja** [comment]
Very nice! 

We're using Mantra and react-komposer. The performance is painful.

We'll test this is out, and let you know of our results. 

@chroz

**Comment by nadeemja** [comment]
Btw, have you yourself tried this in the MantraJS context?

**Comment by maxnowack** [comment]
No, I'm not using Mantra. Please let me know if you're experiencing any issues.
You should also take a look at the [reactive-cache](https://github.com/maxnowack/meteor-reactive-cache) package. It's another package I've build to improve the render performance of [TeamGrid](https://teamgridapp.com)
